src/java.base/share/classes/java/lang/invoke/NativeMethodHandle.java

```diff
@@ -74,11 +88,12 @@ private static boolean hasIllegalType(MethodType type) {
 
     private static boolean isIllegalType(Class<?> pType) {
         return !(pType == long.class
-              || pType == int.class
-              || pType == float.class
-              || pType == double.class
-              || pType == void.class
-              || pType == Object.class);
+                || pType == int.class
+                || pType == float.class
+                || pType == double.class
+                || pType == void.class
+                || pType == MemorySegment.class // Skip checking the argument preset in OpenJ9 for the moment.
+                || pType == SegmentAllocator.class); // Skip checking the argument preset OpenJ9 for the moment.
```

Comment:
Please retain the indentation from upstream.

Reply:
Adjusted as suggested above.